### PR TITLE
DOCK-2105: change GitHub apps log separator to newlines

### PR DESCRIPTION
**Pull request by svonworl**
**Description**
This PR changes the formatting of the github log messages which are based on failed `Validations` so that the submessages are separated by newlines rather than spaces.  In other words, each submessage now appears on its own line.  It also changes the error message from a failed validation associated with a missing primary descriptor file, so the message describes that the file was not found, rather than sort of implying that the missing file itself was missing the primary descriptor. 

An assocated PR will change the UI to format the newlines as newlines. https://github.com/dockstore/dockstore-ui2/pull/1524

Yes, there are repeated strings.  However, amongst the classes in which the string repeated, there didn't seem to be a place that made sense for a constant to reside.  Someday, if we decide to create a class to hold shared error messages and other strings like this, we can put it there.

The motivation for this PR was kathy's review ticket feedback.  

**Issue**
https://ucsc-cgl.atlassian.net/browse/DOCK-2105

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] Check the Snyk dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java [hunk 1]
```diff
@@ -529,25 +529,25 @@ private void addValidationsToLambdaEvent(LambdaEvent event, Collection<WorkflowA
             if (isNotEmpty(validations)) {
                 String subMessage = computeMultiValidationMessage(validations);
                 if (StringUtils.isNotEmpty(subMessage)) {
-                    return String.format("In version '%s' of %s '%s': %s", workflowAndVersion.getVersionName(), workflowAndVersion.getWorkflowType().getTerm(), computeWorkflowName(workflowAndVersion), subMessage);
+                    return String.format("In version '%s' of %s '%s':\n%s", workflowAndVersion.getVersionName(), workflowAndVersion.getWorkflowType().getTerm(), computeWorkflowName(workflowAndVersion), subMessage);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java:532]: Looking at your screenshot in dockstore/dockstore-ui2#1524, I suggest making the first string `"In %s '%s':\n%s"` and getting rid of `workflowAndVersion.getVersionName()` as a parameter -- the version is (sort of, in the reference column) of already there, and if you had multiple workflows with validation errors for the same repository, you would keep seeing the same text again and again, e.g.,

```
In version foo of workflow one:
A validation error
In version foo of workflow two:
Another validation error
In version bar of workflow three:
A validation error
```

**Review by coverbeck — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java (hunk 1 above)
Comment (svonworl) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java:532]: I'm torn here, between your valid concern about the repeating text, and the utility of the version being included in the error message.  What's everyone else think?

Comment (kathy-t) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java:532]: Alternatively, you could mention the version only once before all of the workflow validation messages. Something along the lines of:
```
There are errors in version foo.
In workflow one:
- A validation error
In workflow two:
- Another validation error
```
Re: Charles comment in the dockstore-ui PR about indenting the children, I propose dashes :slightly_smiling_face:

**Review by kathy-t — APPROVED** [review]

**Comment by svonworl** [comment]
Thanks for commenting: it's nice to get solid, well-thought-out feedback from the team!
To me, Kathy's suggestion has the best aesthetics.
Indenting each validation submessage seems to be a generally good thing, I added dashes in front of each.
Regarding the other changes, there's valid arguments for each, and IMHO, balancing all the tradeoffs, there's no clear winner.  So, I'm leaving it as is, with the benefit that if somehow, we someday invoke the formatting code with multiple versions, it still works.

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java (hunk 1 above)
Comment (Richard-Hansen) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java:532]: +1 for dashes and single printout of versions.

**Pull request merged**